backend/src/main/java/com/huellapositiva/application/controller/HandlerPasswordApiController.java

```diff
@@ -76,4 +87,38 @@ public void sendEmailRecovery(@RequestParam("email") String email){
     public void changePassword(@PathVariable("hash") String hash, @RequestParam("newPassword") String password){
         credentialsAction.executePasswordChanging(hash, password);
     }
+
+    @Operation(
+            summary = "Update the password from profile",
+            description = "The user access his profile to update his password.",
+            tags = {"contactPerson, volunteers, profile"}
+    )
+    @ApiResponses(
+            value = {
+                    @ApiResponse(
+                            responseCode = "204",
+                            description = "No content, password updated and confirmation email sent successfully."
+                    ),
+                    @ApiResponse(
+                            responseCode = "409",
+                            description = "Conflict, the old password does not match or the new password is invalid."
+                    ),
+                    @ApiResponse(
+                            responseCode = "500",
+                            description = "Internal server error, could not fetch the user data due to a connectivity issue."
+                    )
+            }
+    )
+    @RolesAllowed({"VOLUNTEER", "CONTACT_PERSON"})
+    @PostMapping("/editPassword")
+    @ResponseStatus(HttpStatus.NO_CONTENT)
+    public void editProfilePassword(@RequestPart("dto") MultipartFile dtoMultipart,
+                                    @AuthenticationPrincipal String email) throws IOException {
+        ChangePasswordDto dto = objectMapper.readValue(dtoMultipart.getBytes(), ChangePasswordDto.class);
```

Comment:
Multipartfile is meant to be used when you are transferring a file not plain json.
See "registerESALAsReviser" method in "ESALApiController" and do not forget to add the required validation annotation to the request DTO.

Reply:
About the validation, you mean this:

![image](https://user-images.githubusercontent.com/59283605/99846893-121a2680-2b6f-11eb-8bcb-573e39923f1f.png)

Reply:
With @NotEmpty should be enough, make sure to create tests for invalid payload as well HTTP 400. 

The Schema annotations are for Swagger documentation although I believe they should not be required.

Reply:
In the test invalid payload (HTTP 400), should i check if the data is well-format (only characteres or numbers and minimun 6 for the length, for example) or do you mean if the DTO has its properties with some data? In the second case, with @AllArgsConstructor is solved.

Reply:
You can perform the validation in the layer you prefer, at controller level or at domain level.

If you want to be consistent with the current approach, perform basic validation like @NotEmpty at controller and leave the rest for the domain logic. Consistency aside, validating as much as you can earliest at the controller level using annotations like @Min, @Size, regex and others is often used to follow the principle of failing fast and ensuring the data handled by the components is safe in terms of being null-free for instance.

@AllArgument does not perform any validation, it just creates a constructor that takes all the properties as parameters.

Reply:
Add the validation